liblwgeom/lwgeom.c

```diff
@@ -2550,15 +2550,13 @@ lwgeom_boundary(LWGEOM *lwgeom)
 
 	switch (lwgeom->type)
 	{
-	case POINTTYPE: {
-		return (LWGEOM *)lwpoint_construct_empty(srid, hasz, hasm);
-	}
+	case POINTTYPE:
 	case MULTIPOINTTYPE: {
-		return (LWGEOM *)lwmpoint_construct_empty(srid, hasz, hasm);
+		return lwgeom_construct_empty(lwgeom->type, srid, hasz, hasm);
 	}
 	case LINETYPE:
 	case CIRCSTRINGTYPE: {
-		if (lwgeom_is_closed(lwgeom))
+		if (lwgeom_is_closed(lwgeom) || lwgeom_is_empty(lwgeom))
```

Comment:
This is where the BoundaryNodeRule would possibly give a different answer (with some BoundaryNodeRule also the closed lines are considered to have a boundary, I think)

Reply:
But I do realize this is out of scope for this work, so feel free to ignore :)